stdlib/asm/collections/mmr.masm

```diff
@@ -104,3 +106,76 @@ export.get
      # stack: [leaf, ...]
   end
 end
+
+#! Load the MMR peak data based on its hash.
+#!
+#! Input: [HASH, mmr_ptr, ...]
+#! Output: [...]
+#!
+#! Where:
+#!  - HASH: is the MMR peak hash, the hash is expected to be padded to an even
+#!    length and to have a minimum size of 16 elements
+#!  - mmt_ptr: the memory location where the MMR data will be written to,
+#!    starting with the MMR forest (its total leaves count) followed by its peaks
+#!
+#! Cycles: 159 + 9 extra_peak_pair cycles
+#!    where `extra_peak` is the number of peak pairs in addition to the first
+#!    16, i.e. `round_up((num_of_peaks - 16) / 2)`
+export.unhash
+  # load the peaks to the advice_stack (0 cycles)
+  adv.keyval
+  # operand_stack => [HASH, mmr_ptr, ...]
+  # advice_stack => [peaks*, ...]
+
+  # load the number of leaves in the MMR to the advice_stack, the number of
+  # leaves is defined by the hash of the MMR peaks + 1 (3 cycles)
+  add.1 adv.keyval sub.1
+  # operand_stack => [HASH, mmr_ptr, ...]
+  # advice_stack => [forest, peaks*, ...]
+
+  # load the size from the advice stack (1 cycles)
+  adv_push.1
+  # operand_stack => [forest, HASH, mmr_ptr, ...]
+  # advice_stack => [peaks*, ...]
+
+  # save the forest to memory (4 cycles)
+  dup dup.6 mem_store
+  # => [forest, HASH, mmr_ptr, ...]
+
+  # count number of peaks (69 cycles)
+  u32split u32unchecked_popcnt swap u32unchecked_popcnt add
+  # => [count, HASH, mmr_ptr, ...]
+
+  # the peaks are padded to a minimum length of 16 (10 cycles)
+  push.16 u32unchecked_max
+  # => [count_min, HASH, mmr_ptr, ...]
```

Comment:
`u32unchecked_max` should be correct. Here we are not comparing against the `number_of_leaves`, but against the `number_of_peaks`. Because we use a `u64` to represent the number of leaves, and the number of peaks is equal to the number of `true` bits in `number_of_leaves`, the maximum value we will have here is `64` which is `u32`

Reply:
My concern was more about whether we can guarantee that `count` is actually a `u32` value - but looking at how it is computed on line 146, it should be fine (seems like `count` will never be higher than 63).